### PR TITLE
[chassis] Minigraph parser changes

### PR DESCRIPTION
**Pull request by arlakshm**
#### Why I did it
Update the minigraph parser to support chassis production minigraph

##### Work item tracking
- Microsoft ADO **28223907**:

#### How I did it
Make changes the following changes in the parser to parse production minigraph for sonic T2 chassis and generate the configuration required for SONiC T2

- Support parsing of the `system-ports` for voq chassis from the production minigraph
- Generate the chassis level port alias for all the ports in the chassis from the minigraph
- Generate the chassis hostname in the device metadata for all the linecards from the minigraph
- Generate the `voqinband` interface for voq chassis.
- Update the actual port speed for linecards based on the minigraph.
- Add UT for voq and packet chassis minigraphs.

#### How to verify it
UT and test on sonic chassis

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

File: src/sonic-config-engine/minigraph.py [hunk 1]
```diff
@@ -1685,27 +2168,49 @@ def parse_xml(filename, platform=None, port_config_file=None, asic_name=None, hw
...
     if len(system_defaults) > 0:
         results['SYSTEM_DEFAULTS'] = system_defaults
+   
+    if asic_name is not None:
+        results['DEVICE_METADATA']['localhost']['asic_name'] =  asic_name
+    
+    # for single asic Voq Linecards the asic_name needs to populated to "Asic0"
+    if switch_type == "voq" or chassis_type in [CHASSIS_CARD_VOQ]:
+        if not is_multi_asic():
+            results['DEVICE_METADATA']['localhost']['asic_name'] =  "Asic0"
```
Comment (judyjoseph) [inline src/sonic-config-engine/minigraph.py:2178]: Does it matter if it is 'Asic0' or' ASIC0' - remember there was an issue earlier

**Review by judyjoseph — APPROVED** [review]

File: src/sonic-config-engine/minigraph.py (hunk 1 above)
Comment (arlakshm) [reply src/sonic-config-engine/minigraph.py:2178]: This is 'Asic0'

**Comment by lguohan** [comment]
@arlakshm , can you sync with qi and xincun on this, this would impact how ndm generate the golden config for T2.

**Pull request merged**